opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -694,27 +694,23 @@ def start_as_current_span(
     def start_span(  # pylint: disable=too-many-locals
         self,
         name: str,
-        parent: trace_api.ParentSpan = trace_api.Tracer.CURRENT_SPAN,
+        parent: Optional[context_api.Context] = None,
         kind: trace_api.SpanKind = trace_api.SpanKind.INTERNAL,
         attributes: types.Attributes = None,
         links: Sequence[trace_api.Link] = (),
         start_time: Optional[int] = None,
         set_status_on_exception: bool = True,
     ) -> trace_api.Span:
-        if parent is Tracer.CURRENT_SPAN:
-            parent = trace_api.get_current_span()
 
-        parent_context = parent
-        if isinstance(parent_context, trace_api.Span):
-            parent_context = parent.get_context()
+        parent_context = trace_api.get_current_span(parent).get_context()
```

Comment:
Now that there is the concept of a `Context` being passed into span start, I'm wondering if it is valuable to rename `span.get_context()` to `span.get_span_context()` to avoid confusion. I know that the `Context` isn't actually a PART of `Span`, it probably is only here (during start and use) that it would be confusion. Thoughts?

Reply:
And maybe even renaming the variable from `parent_context` to `parent_span_context`.

Reply:
I would +1 on "get_span_context", although with https://github.com/open-telemetry/opentelemetry-specification/pull/1033 that might not even be necessary.

Reply:
I renamed `get_context` to `get_span_context` and `parent_context` to `parent_span_context` where it made sense. I tested the water with renaming `context` in the Span to `span_context` and it turned out to be pretty messy. Not sure if it's worth the rename on that one. Thoughts @lzchen & @toumorokoshi?

Reply:
I think that's fine, I don't know if I have a strong opinion on the attribute name on the span.

And I'm completely comfortable with taking these PRs piecemeal too.